vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,185 @@
+     * Specifies the password of the communication user
+     */
+    'password'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
+ *
+ * !!! note "Git Repository and Software Component"
+ *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
+ *       Currently, those two names are used synonymous.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        ConfigurationHelper configHelper = ConfigurationHelper.newInstance(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+
+        Map configuration = configHelper.use()
+
+        configHelper
+            .withMandatoryProperty('host', 'Host not provided')
+            .withMandatoryProperty('repositoryName', 'Repository / Software Component not provided')
+            .withMandatoryProperty('username')
+            .withMandatoryProperty('password')
+
+        String usernameColonPassword = configuration.username + ":" + configuration.password
+        String authToken = usernameColonPassword.bytes.encodeBase64().toString()
+        String urlString = configuration.host + '/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull'
+        echo "[${STEP_NAME}] General Parameters: URL = \"${urlString}\", repositoryName = \"${configuration.repositoryName}\""
+
+        String urlPullEntity = triggerPull(configuration, urlString, authToken)
+
+        if (urlPullEntity != null) {
+            String finalStatus = pollPullStatus(urlPullEntity, authToken)
+            if (finalStatus != 'S') {
+                error "[${STEP_NAME}] Pull Failed"
+            }
+        }
+    }
+}
+
+private String triggerPull(Map configuration, String url, String authToken) {
+
+    String entityUri = null
+    String headerFile = "headerAuth.txt"
+
+    def xCsrfTokenScript = """#!/bin/bash
+        curl -I -X GET ${url} \
+        -H 'Authorization: Basic ${authToken}' \
+        -H 'Accept: application/json' \
+        -H 'x-csrf-token: fetch' \
+        -D ${headerFile} \
+        | awk 'BEGIN {FS=": "}/^x-csrf-token/{print \$2}'
+    """
+
+    def xCsrfToken = sh (
+        script : xCsrfTokenScript,
+        returnStdout: true )
+
+    checkRequestStatus(headerFile)
+
+    String headerFilePost = "headerPost.txt"
+    def scriptPull = """#!/bin/bash
+        curl -X POST \"${url}\" \
+        -H 'Authorization: Basic ${authToken}' \
+        -H 'Accept: application/json' \
+        -H 'Content-Type: application/json' \
+        -H 'x-csrf-token: ${xCsrfToken.trim()}' \
```

Comment:
I would like to recommand to `trim()` above: `sh(./.).trim()`

Reply:
trim() is not necessary anymore